tests/parser/syntax/test_unbalanced_return.py

```diff
@@ -0,0 +1,106 @@
+import pytest
+from pytest import (
+    raises,
+)
+
+from vyper import (
+    compiler,
+)
+from vyper.exceptions import (
+    StructureException,
+)
+
+fail_list = [
+    """
+@public
+def foo() -> int128:
+    pass
+    """,
+    """
+@public
+def foo() -> int128:
+    if False:
+        return 123
+    """,
+    """
+@public
+def test() -> int128:
+    if 1 == 1 :
+        return 1
+        if True:
+            return 0
+    else:
+        assert False
+    """,
+    """
+@private
+def valid_address(sender: address) -> bool:
+    selfdestruct(sender)
+    return True
+    """
+]
+
+
+@pytest.mark.parametrize('bad_code', fail_list)
+def test_missing_return(bad_code):
+    with raises(StructureException):
+        compiler.compile_code(bad_code)
+
+
+valid_list = [
+    """
+@public
+def foo() -> int128:
+    return 123
+    """,
+    """
+@public
+def foo() -> int128:
+    if True:
+        return 123
+    else:
+        raise "test"
+    """,
+    """
+@public
+def foo() -> int128:
+    if False:
+        return 123
+    else:
+        selfdestruct(msg.sender)
```

Comment:
Still feel like this should raise a compiler exception since it would silently fail in this case.

Compounded by a dead code block above... so I'm not sure what the intent of this test case is.

Reply:
@charles-cooper  I think #1390 would address this correct?

Reply:
@jacqueswww yeah, that's the idea for #1390. Out of scope of this PR, I think.

@fubuloubu The purpose of this test is to show that `return` and `selfdestruct` behave the same for the return checker.